initrd/utils.go

```diff
@@ -55,3 +61,147 @@ func compressFiles(output string, writer *cpio.Writer, reader *os.File) error {
 
 	return nil
 }
+
+func convertToErofs(output string, source string, isTar, allRoot bool) error {
+	// Extract the archive to a temporary directory
```

Comment:
It looks like the large `if isTar` code path is reversing the earlier change you were doing with `Create` and changing this into a `string` input.  Given the previous suggestion this should no longer be necessary and the `if isTar` should be simpler.

However, I am in favour of totally removing this function and replacing in-line `erofs.Create` where this method is called (+ preamble).